packages/ui-scripts/lib/build/babel.js

```diff
@@ -29,83 +29,89 @@ import { fileURLToPath } from 'url'
-    process.env
 
-  const args = process.argv.slice(2)
-
-  // positional: ui-build src --watch
-  const firstArg = args[0]
-  const src = firstArg && firstArg.indexOf('--') < 0 ? firstArg : 'src'
-
-  // uncomment the extensions arg after renaming the files from js -> ts happens
-  let babelArgs = ['--extensions', '.ts,.tsx,.js,.jsx']
-
-  if (args.includes('--copy-files')) {
-    babelArgs.push('--copy-files')
-  }
-
-  babelArgs = babelArgs.concat([
-    src,
-    '--ignore "src/**/*.test.js","src/**/__tests__/**"'
-  ])
-
-  let envVars = [
-    OMIT_INSTUI_DEPRECATION_WARNINGS
-      ? `OMIT_INSTUI_DEPRECATION_WARNINGS=1`
-      : false
-  ]
-
-  if (args.includes('--watch')) {
-    envVars = envVars.concat(['NODE_ENV=development']).filter(Boolean)
-    babelArgs.push('--watch')
-  } else {
-    envVars = envVars
-      .concat([
-        `NODE_ENV=${BABEL_ENV || NODE_ENV || 'production'}`,
-        DEBUG ? `DEBUG=1` : false
-      ])
-      .filter(Boolean)
-  }
+export default {
+  command: 'build',
+  desc: '',
+  builder: (yargs) => {
+    yargs.option('copy-files', { boolean: true, desc: '' })
+    yargs.option('watch', { boolean: true, desc: '' })
+    yargs.option('modules', { string: true, desc: '' })
+    yargs.check((argv) => {
+      if (
+        argv.modules &&
+        argv.modules.split(',').some((arg) => !['es', 'cjs'].includes(arg))
+      ) {
+        throw new Error(`Invalid --modules argument: '${argv.modules}'`)
+      }
+      return true
+    })
+    yargs.strictOptions(true)
+  },
+  handler: async (argv) => {
+    const { BABEL_ENV, NODE_ENV, DEBUG, OMIT_INSTUI_DEPRECATION_WARNINGS } =
+      process.env
+
+    // positional: ui-scripts build src --watch
+    const src = argv._[1] || 'src'
+
+    // uncomment the extensions arg after renaming the files from js -> ts happens
+    let babelArgs = ['--extensions', '.ts,.tsx,.js,.jsx']
+
+    if (argv.copyFiles) {
+      babelArgs.push('--copy-files')
+    }
 
-  let modules = ['es']
+    babelArgs = babelArgs.concat([
+      src,
+      '--ignore "src/**/*.test.js","src/**/__tests__/**"'
+    ])
 
-  if (args.includes('--modules')) {
-    //  eslint-disable-next-line no-unused-vars
-    const [_, arg] = args.splice(args.indexOf('--modules'), 2)
+    let envVars = [
+      OMIT_INSTUI_DEPRECATION_WARNINGS
+        ? `OMIT_INSTUI_DEPRECATION_WARNINGS=1`
+        : false
+    ]
 
-    if (!arg) {
-      throw new Error('Missing --modules argument')
+    if (argv.watch) {
+      envVars = envVars.concat(['NODE_ENV=development']).filter(Boolean)
+      babelArgs.push('--watch')
+    } else {
+      envVars = envVars
+        .concat([
+          `NODE_ENV=${BABEL_ENV || NODE_ENV || 'production'}`,
+          DEBUG ? `DEBUG=1` : false
+        ])
+        .filter(Boolean)
     }
 
-    modules = arg.split(',')
+    let modules = ['es']
 
-    if (modules.some((mod) => !['es', 'cjs'].includes(mod))) {
-      throw new Error(`Invalid --modules argument: '${arg}'`)
```

Comment:
I think the original solution is better here: Only allow `es` or `cjs` as values and error out when the `modules` argument is missing. You could use http://yargs.js.org/docs/#api-reference-choiceskey-choices and http://yargs.js.org/docs/#demandOption

Reply:
I think the original solution defaulted to `es` if there were no `--modules` argument. Should I keep it like that or use `demandOptions`? I will also add the yargs `choices` option

Reply:
oh you're right, it had some wonky double check for the `modules` argument. We can default to `es`, there is also a `yargs` option for default :)